src/pages/[platform]/build-a-backend/auth/sign-in-custom-flow/index.mdx

```diff
@@ -0,0 +1,1005 @@
+        if case .confirmSignInWithCustomChallenge(_) = signInResult.nextStep {
+            // Ask the user to enter the custom challenge.
+        } else {
+            print("Sign in succeeded")
+        }
+    } catch let error as AuthError {
+        print("Sign in failed \(error)")
+    } catch {
+        print("Unexpected error: \(error)")
+    }
+}
+```
+
+</Block>
+
+<Block name="Combine">
+
+```swift
+func signIn(username: String) -> AnyCancellable {
+    Amplify.Publisher.create {
+        let options = AWSAuthSignInOptions(authFlowType: .customWithoutSRP)
+        try await Amplify.Auth.signIn(username: username,
+                                    options: .init(pluginOptions: options))
+    }.sink {
+        if case let .failure(authError) = $0 {
+            print("Sign in failed \(authError)")
+        }
+    }
+    receiveValue: { result in
+        if case .confirmSignInWithCustomChallenge(_) = result.nextStep {
+            // Ask the user to enter the custom challenge.
+        } else {
+            print("Sign in succeeded")
+        }
+    }
+}
+```
+
+</Block>
+
+</BlockSwitcher>
+
+Since this is a custom authentication flow with a challenge, the result of the signin process has a next step `.confirmSignInWithCustomChallenge`. Implement a UI to allow the user to enter the custom challenge.
+
+## Confirm sign in with custom challenge
+
+Get the custom challenge (`1234` in this case) from the user and pass it to the `confirmSignin()` api.
+
+<BlockSwitcher>
+
+<Block name="Async/Await">
+
+```swift
+func customChallenge(response: String) async {
+    do {
+      _ = try await Amplify.Auth.confirmSignIn(challengeResponse: response)
+      print("Confirm sign in succeeded") 
+    } catch let error as AuthError {
+      print("Confirm sign in failed \(error)")
+    } catch {
+      print("Unexpected error: \(error)")
+    }
+}
+```
+
+</Block>
+
+<Block name="Combine">
+
+```swift
+func customChallenge(response: String) -> AnyCancellable {
+    Amplify.Publisher.create {
+        try await Amplify.Auth.confirmSignIn(challengeResponse: response)
+        }.sink {
+            if case let .failure(authError) = $0 {
+                print("Confirm sign in failed \(authError)")
+            }
+        }
+        receiveValue: { _ in
+            print("Confirm sign in succeeded")
+        }
+}
+```
+
+</Block>
+
+</BlockSwitcher>
+
+You will know the sign in flow is complete if you see the following in your console window:
+
+```console
+Confirm sign in succeeded
+```
+
+### Lambda Trigger Setup
+
+AWS Amplify now supports creating functions as part of the AWS Amplify. For more information on the Functions and how to start with them check out [Functions documentation](/[platform]/build-a-backend/functions/). In addition, more information on available triggers can be found in the [Cognito documentation](https://docs.aws.amazon.com/cognito/latest/developerguide/user-pool-lambda-challenge.html).
```

Comment:
AWS Amplify now supports creating functions as part of the AWS Amplify I think this sentence may be off? Maybe we can just cut the sentence

Reply:
Yes it is weird. However I think it requires a bridge sentence. What do you think of the following:

AWS Amplify now supports creating functions as part of its new backend experience.